streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -17,242 +17,62 @@
-         */
-        @Override
-        public byte[] serialize(Object data) {
-            try {
-                return ((String) data).getBytes("UTF-8");
-            } catch (UnsupportedEncodingException e) {
-                throw new AssertionError(e);
-            }
-        }
+    private final int replicationFactor;
+    private StreamsKafkaClient streamsKafkaClient;
 
-        /**
-         * @throws AssertionError if the byte String encoding type is not supported
-         */
-        @Override
-        public Object deserialize(byte[] bytes) {
-            try {
-                if (bytes == null)
-                    return null;
-                else
-                    return new String(bytes, "UTF-8");
-            } catch (UnsupportedEncodingException e) {
-                throw new AssertionError(e);
-            }
-        }
-    }
 
     public InternalTopicManager() {
-        this.zkClient = null;
         this.replicationFactor = 0;
         this.windowChangeLogAdditionalRetention = WINDOW_CHANGE_LOG_ADDITIONAL_RETENTION_DEFAULT;
     }
-
-    public InternalTopicManager(String zkConnect, final int replicationFactor, long windowChangeLogAdditionalRetention) {
-        this.zkClient = new ZkClient(zkConnect, 30 * 1000, 30 * 1000, new ZKStringSerializer());
+    
+    public InternalTopicManager(StreamsKafkaClient streamsKafkaClient, final int replicationFactor, long windowChangeLogAdditionalRetention) {
+        this.streamsKafkaClient = streamsKafkaClient;
         this.replicationFactor = replicationFactor;
         this.windowChangeLogAdditionalRetention = windowChangeLogAdditionalRetention;
     }
 
-    public void makeReady(InternalTopicConfig topic, int numPartitions) {
-        boolean topicNotReady = true;
-
-        while (topicNotReady) {
-            Map<Integer, List<Integer>> topicMetadata = getTopicMetadata(topic.name());
-
-            if (topicMetadata == null) {
-                try {
-                    createTopic(topic, numPartitions, replicationFactor);
-                } catch (ZkNodeExistsException e) {
-                    // ignore and continue
+    /**
+     * If the topic exists do nothing (we don't change partititions or delete the existing topics).
+     * Otherwise create the new topic.
+     * @param topic
+     * @param numPartitions
+     */
+    public void makeReady(final InternalTopicConfig topic, final int numPartitions) {
+
+        if (!streamsKafkaClient.topicExists(topic.name())) {
+            streamsKafkaClient.createTopic(topic, numPartitions, replicationFactor, windowChangeLogAdditionalRetention);
+        } else {
+            final MetadataResponse.TopicMetadata topicMetadata = streamsKafkaClient.getTopicMetadata(topic.name());
+            if (topicMetadata != null) {
+                if (topicMetadata.error().code() != 0) {
+                    throw new StreamsException("Topic metadata request returned with error code " + topicMetadata.error().code());
```

Comment:
Couldn't find it in the docs. In the code we have "org.apache.kafka.common.protocol.Errors". I can also print message() or exceptionName() instead of the code. Adding the message().